CMakeLists.txt

```
@@ -602,11 +612,57 @@ if (YUZU_USE_BUNDLED_FFMPEG)
                 --enable-hwaccel=h264_vaapi
                 --enable-hwaccel=vp9_vaapi
                 --enable-libdrm)
+            list(APPEND FFmpeg_HWACCEL_INCLUDE_DIRS
+                ${LIBDRM_INCLUDE_DIRS}
+                ${X11_INCLUDE_DIRS}
+                ${LIBVA-DRM_INCLUDE_DIRS}
+                ${LIBVA-X11_INCLUDE_DIRS}
+                ${LIBVA_INCLUDE_DIRS}
+            )
             message(STATUS "VA-API found")
         else()
             set(FFmpeg_HWACCEL_FLAGS --disable-vaapi)
         endif()
 
+        if (FFNVCODEC_FOUND AND CUDA_FOUND)
+            list(APPEND FFmpeg_HWACCEL_FLAGS
+                --enable-cuvid
+                --enable-ffnvcodec
+                --enable-nvdec
+                --enable-cuda-llvm
+                --enable-hwaccel=h264_nvdec
+                --enable-hwaccel=vp9_nvdec
+                --extra-cflags=-I${CUDA_INCLUDE_DIRS}
+            )
```

```suggestion
           list(APPEND FFmpeg_HWACCEL_FLAGS
                --enable-nonfree
                --enable-cuvid
                --enable-ffnvcodec
                --enable-nvdec
                --enable-cuda-nvcc
                --enable-hwaccel=h264_nvdec
                --enable-hwaccel=vp9_nvdec
                --extra-cflags=-I${CUDA_INCLUDE_DIRS}
                --nvccflags='-gencode arch=compute_52,code=sm_52 -O2'
            )
```

This is what I had to do to get a successful build on GCC 11.1 Ubuntu 20.10
I believe cuda_llvm is just clang builds.
the default nvcc flags are found [here.](https://github.com/FFmpeg/FFmpeg/blob/115f5e803502451ac64fa698730923299e5acc4e/configure#L4376)

[This explains the different gencodes.](https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/)

---

We cannot set `--enable-nonfree` -- see https://www.ffmpeg.org/legal.html part 1.

We should just axe `--enable-cuda-*` since it doesn't seem to make a difference in yuzu on my end.

I don't understand why we need to specify Maxwell v2 when the FFmpeg's flags default to Kepler.

Maybe unrelated, or maybe it is related since it could affect your ability to build yuzu, but I recommend you use a supported operating system. See https://wiki.ubuntu.com/Releases

---

I was following [this guide](https://docs.nvidia.com/video-technologies/video-codec-sdk/ffmpeg-with-nvidia-gpu/)  and tried to compile without the nonfree flag but it complained that nvcc required it so that's sad.

The gencode I used is specific for my card (GTX 980) but the page I linked explains that the cuda toolkit 11 supports up to maxwell and that it will be deprecated in 12. The article also explains that it is a good idea to target the arch and gencode for the gpu it will run on for performance reasons.

I am actually on 21.04 but my brain is used to typing 20.10.... sorry about that

---

```cmake
list(APPEND FFmpeg_HWACCEL_FLAGS
                --enable-cuvid
                --enable-ffnvcodec
                --enable-nvdec
                --enable-hwaccel=h264_nvdec
                --enable-hwaccel=vp9_nvdec
                --extra-cflags=-I${CUDA_INCLUDE_DIRS}
            )
```

Tested with success, what a relief.
Thanks for the heads up on the nonfree problem.